libexec/pyenv-exec

```diff
@@ -41,7 +41,7 @@ for script in "${scripts[@]}"; do
 done
 
 shift 1
-# CPython's `sys.executable` requires the `PYENV_BIN_PATH` to be at the top of the `PATH`.
-# https://github.com/pyenv/pyenv/issues/98
-export PATH="${PYENV_BIN_PATH}:${PATH}"
-exec -a "$PYENV_COMMAND" "$PYENV_COMMAND_PATH" "$@"
+if [ "$PYENV_VERSION" != "system" ]; then
```

Comment:
Maybe I'm missing spme context, but how it should behave if there are multiple versions in PYENV_VERSION and the "system" appeared in the middle...?

Reply:
Good point.

It should only be added if the command is from a non-system version after all, i.e. with `PYENV_VERSION=system:3.5.3`, PATH should not be changed for `pip`, which is used from `system` then.
To make this work pyenv-which needs to return the used version in some new mode then maybe?

Reply:
Could we go ahead with this, which I think fixes the common case of `pyenv system` unfortunately prepending `/usr/bin` to PATH and breaking other software not expecting `/usr/bin` to be so early in PATH?
If not, is there another solution to deal with this?

Reply:
I've changed it now to get the used version out of `pyenv-which`.